### PR TITLE
Update statsd exporter library

### PR DESCRIPTION
**Pull request by sagikazarmark**
This PR updates the prometheus statsd library that in turn removes tons of transitive dependencies.

See prometheus/common#255 for context.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -2,8 +2,8 @@ module contrib.go.opencensus.io/exporter/prometheus
 
 require (
 	github.com/google/go-cmp v0.5.5
-	github.com/prometheus/client_golang v1.9.0
-	github.com/prometheus/statsd_exporter v0.20.0
+	github.com/prometheus/client_golang v1.10.1-0.20210603120351-253906201bda
```
Comment (dashpole) [inline go.mod]: do you know how often prometheus cuts releases?  Should we wait for a tagged version?

Comment (sagikazarmark) [reply go.mod]: Yes, that probably makes sense. AFAIK they are going to cut a release soon: prometheus/client_golang#876

Comment (sagikazarmark) [reply go.mod]: Updated to the next release

**Comment by BradErz** [comment]
Sorry to be a pain in the ass but is there any chance of getting this merged?

**Review by bogdandrutu — APPROVED** [review]

**Review by jsuereth — APPROVED** [review]

**Comment by sagikazarmark** [comment]
Thanks! Any chance to get the merged changes released?

**Pull request merged**